File: src/dataset/javascript/resistor-color-duo_exercism-javascript-programmiri/openAI/ChatGPT4_resistor-color-duo_v2.js
const colorMap = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};

function value(colorNameList) {
  let result = 0;
  for (const color of colorNameList) {
    result = result * 10 + colorMap[color];
  }
  return result;
}

export { value };

File: src/dataset/go/resistor_color_duo_Exercism_go_thinkverse/openAI/ChatGPT4_resistor_color_duo_v2.go
package resistorcolorduo

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	if len(colors) < 2 {
		return 0
	}

	colorValues := [...]int{
		'k': 0, // black
		'b': 1, // brown
		'r': 2, // red
		'o': 3, // orange
		'y': 4, // yellow
		'g': 5, // green
		'u': 6, // blue
		'v': 7, // violet
		'e': 8, // grey
		'w': 9, // white
	}

	return (colorValues[colors[0][0]] * 10) + colorValues[colors[1][0]]
}